Add facets/absolute/custom to excel and plot

inodb opened the pull request:
- Get cancer_cell_frac Pr_somatic_clonal and ccf_CI95_low from absolute results.
- Get facets LOH from facets/geneCN.em.reviewed.txt. Basically the original
  facets file after reviewing
- Get whatever other type of annotations for the mutations from
  summary/annotation.tsv merge on TUMOR_SAMPLE CHROM POS REF ALT columns
- Add MUTATION_SUMMARY sheet in excel that lists all NS and S mutations. Remove
  annotations that are not NS and S from | separated columns
- Add likely_pathogenic
  - for snv: fathmm_pred is "CANCER" or chasm score < 0.3
  - for indel: dbNSFP_PROVEAN_PRED is D or dbNSFP_MutationTaster_pred is D
- Add cancer_gene
      - either cancer_gene_census, kandoth or lawrence is true
- Output all excel sheets separately in `summary/tsv`
- Plot heatmaps for NS mutations in all samples using the excel file
  - ccf from absolute
  - LOH from facets
  - likely_pathogenic
  - cancer_gene

inodb committed:
Add facets/absolute/custom to excel and plot

- Get cancer_cell_frac Pr_somatic_clonal and ccf_CI95_low from absolute results.
- Get facets LOH from facets/geneCN.em.reviewed.txt. Basically the original
  facets file after reviewing
- Get whatever other type of annotations for the mutations from
  summary/annotation.tsv merge on TUMOR_SAMPLE CHROM POS REF ALT columns
- Add MUTATION_SUMMARY sheet in excel that lists all NS and S mutations. Remove
  annotations that are not NS and S from | separated columns
- Add likely_pathogenic
        - for snv: fathmm_pred is "CANCER" and chasm score > 0.3
        - for indel: dbNSFP_PROVEAN_PRED is D or dbNSFP_MutationTaster_pred is
          D
- Add cancer_gene
        - either cancer_gene_census, kandoth or lawrence is true
- Plot heatmaps for NS mutations in all samples using the excel file
        - ccf from absolute
        - LOH from facets
        - likely_pathogenic
        - cancer_gene

hxrts commented:
huge commit. I don't have a good test set for this just now but will in a day or so.

inodb commented:
This should btw also fix #83 and #41

inodb committed:
Mutation summary add optional ExAC_AF threshold

In case of a bad sample, it can be useful to set a threshold on the ExAC_AF
column. This adds a config variable EXCEL_MAX_EXAC_AF to set the max value for
mutations in the mutation_summary excel sheet, it is 1 by default.

inodb committed:
Fix CHASM prediction inversion
